src/main/java/org/aksw/simba/squirrel/sink/impl/sparql/SparqlBasedSink.java

```diff
@@ -0,0 +1,82 @@
+package org.aksw.simba.squirrel.sink.impl.sparql;
+
+import org.aksw.simba.squirrel.data.uri.CrawleableUri;
+import org.aksw.simba.squirrel.sink.Sink;
+import org.apache.jena.graph.Triple;
+import org.apache.jena.update.UpdateExecutionFactory;
+import org.apache.jena.update.UpdateFactory;
+import org.apache.jena.update.UpdateProcessor;
+import org.apache.jena.update.UpdateRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class SparqlBasedSink implements Sink {
+
+    public static final String JENA_PORTS_KEY = "JENA_PORT";
+    public static final String JENA_CONTAINER_NAME_KEY = "JENA_HOST_NAME";
+    private static final int SENDING_INTERVAL_BUFFERED_LIST = 100;
+    private static String strContentDatasetUriUpdate;
+    private static String strContentDatasetUriQuery;
+    private List<Triple> lstBufferedTriples;
+
+    @SuppressWarnings("unused")
+    private static final Logger LOGGER = LoggerFactory.getLogger(SparqlBasedSink.class);
+
+
+    public SparqlBasedSink() throws Exception {
+        Map<String, String> env = System.getenv();
+        String containerName = null;
+        String port = null;
+        if (env.containsKey(JENA_CONTAINER_NAME_KEY) || env.containsKey(JENA_PORTS_KEY)) {
+            containerName = env.get(JENA_CONTAINER_NAME_KEY);
+            port = env.get(JENA_PORTS_KEY);
+        } else {
+            String msg = "Couldn't get " + JENA_CONTAINER_NAME_KEY + " or " + JENA_PORTS_KEY + " from the environment.";
+            throw new Exception(msg);
+        }
+
+        String datasetPrefix = "http://" + containerName + ":" + port + "/ContentSet/";
+        strContentDatasetUriUpdate = datasetPrefix + "update";
+        strContentDatasetUriQuery = datasetPrefix + "query";
+    }
+
+    public void addMetadata() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void addTriple(CrawleableUri uri, Triple triple) {
+        lstBufferedTriples.add(triple);
+        if (lstBufferedTriples.size() >= SENDING_INTERVAL_BUFFERED_LIST) {
+            sendAllTriplesToDB(uri);
+        }
+    }
+
+    @Override
+    public void openSinkForUri(CrawleableUri uri) {
+        lstBufferedTriples = new ArrayList<>(SENDING_INTERVAL_BUFFERED_LIST);
+    }
```

Comment:
Your buffer is not thread safe but we may want to write to more than one graph (or more than one uri) at the same time :wink: 

You might want to take a look at the file base sink which is already thread safe (especially at the stream mapping which https://github.com/dice-group/Squirrel/blob/master/src/main/java/org/aksw/simba/squirrel/sink/impl/file/FileBasedSink.java)

Reply:
I never worked with this `ConcurrentHashMap` and I am not sure whether it will do the trick. However, it should be fine for now :thinking: